Sources/DNSResolver.swift

```diff
@@ -55,9 +55,9 @@ final class DNSResolver {
                                                 retain: nil, release: nil, copyDescription: kCopyNoOperation)
 
         let hostReference = CFHostCreateWithName(kCFAllocatorDefault, host as CFString).takeUnretainedValue()
-        resolver.timer = Timer.scheduledTimer(timeInterval: timeout, target: resolver,
-                                              selector: #selector(DNSResolver.onTimeout),
-                                              userInfo: hostReference, repeats: false)
+        resolver.timer = .scheduledTimer(timeInterval: timeout, target: resolver,
+                                         selector: #selector(onTimeout),
```

Comment:
I intentionally use the full selector names to avoid ambiguities when reading it.

Reply:
Full selector => we are calling a selector of a potentially different class 
Short selector => we are calling a selector that definitely belongs to this class

Reply:
It's a good argument but I don't think it holds on this particular example (or any that takes a `target`) since the type that maters is `resolver` not `self`. So you don't actually get any compiler guarantee (you could use a diff class for `resolver`).

I don't feel really strong about this one. Just not a big fan of the inconsistency when the member is, in fact, on a different class.